src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -152,17 +166,14 @@ def evaluate(
                 )
 
                 valid_indices_for_period = np.nonzero(is_operational_setting_valid.values)[0]
-                if len(valid_indices_for_period) == 0:
```

Comment:
this was here for a reason..I guess. did you check something before you removed it?